dotCMS/src/main/java/com/dotcms/publishing/job/SiteSearchJobImpl.java

```diff
@@ -2,203 +2,280 @@
-                }
-
-            }
-
-            config.setHosts( hosts);
-
-            // reuse or create new indexes as needed
-            String indexAlias = dataMap.getString("indexAlias");
-            ESIndexAPI iapi=new ESIndexAPI();
-            Map<String,String> aliasMap=iapi.getAliasToIndexMap(APILocator.getSiteSearchAPI().listIndices());
-
-            if(counter==1){
-	            if(indexAlias.equals(APILocator.getIndiciesAPI().loadIndicies().getSiteSearch())){
-	            	Logger.info(this, "Index Alias is DEFAULT");
-	                indexName = APILocator.getIndiciesAPI().loadIndicies().getSiteSearch();
-	            }
-	            else if(indexAlias.equals("create-new")){
-	            	Logger.info(this, "Index Alias is default");
-	            	createNew=true;
-	                indexName = SiteSearchAPI.ES_SITE_SEARCH_NAME  + "_" + ESMappingAPIImpl.datetimeFormat.format(new Date());
-	                APILocator.getSiteSearchAPI().createSiteSearchIndex(indexName, null, 1);
-	            }
-	            else {
-	                indexName=aliasMap.get(indexAlias);
-	                Logger.info(this, "Index Alias is " + indexName);
-	            }
-            }
-            if(createNew && languageToIndex.length == counter){
-            	config.setSwitchIndexWhenDone(true);
-            }
-
+            config.setJobName(jobName);
+            config.setHosts(hosts);
+            config.setNewIndexName(newIndexName);
             config.setIndexName(indexName);
-
-
-
-            // if it is going to be an incremental job, write the bundle to the same folder
-            // every time.  Otherwise, create a new folder using a date stamp.
-            if(dataMap.get("incremental")!=null){
-                config.setId(StringUtils.camelCaseLower(config.getJobName()));
-            }
-            else{
-                String x = UtilMethods.dateToJDBC(new Date()).replace(':', '-').replace(' ', '_');
-                config.setId(x);
-            }
-
+            config.setId(bundleId);
             config.setStartDate(startDate);
             config.setEndDate(endDate);
             config.setIncremental(incremental);
             config.setUser(userToRun);
+
             if(include) {
                 config.setIncludePatterns(paths);
-            }
-            else {
+            } else {
                 config.setExcludePatterns(paths);
             }
 
-            APILocator.getPublisherAPI().publish(config,status);
-
+            //We should always replace the index when performing on non-incremental mode.
+            //That means we drop the old one and re-use the alias.
+            //But we only activate the new index when the old one was the default.
+            //Or there wasn't any previous index.
+            //it must be done on the last round of our loop.
+            boolean switchIndex = !incremental && !listIterator.hasNext();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'switchIndex' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093862)